### DISCUSSION TITLE
Can't extract gradle dependencies due to Java version

### DISCUSSION THREAD

**Discussion by Aurelien30000** [renovatebot/renovate#10767] [Request Help] [answered]
**How are you running Renovate?**

- [x] WhiteSource Renovate hosted app on github.com
- [ ] Self hosted

**Describe the bug**

My project needs Java 16 to build and run correctly. However, trying to extract gradle dependencies and building the project, renovatebot fails because it uses Java 11, and there is no way to update/change it.

Thanks!

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

```
WARN: Gradle extraction failed
{
  "errMessage": "Command failed: docker run --rm --name=renovate_gradle --label=renovate_child -v \"/mnt/renovate/gh/DomiCraftFR/DomiDependencies\":\"/mnt/renovate/gh/DomiCraftFR/DomiDependencies\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -e GRADLE_OPTS -w \"/mnt/renovate/gh/DomiCraftFR/DomiDependencies\" docker.io/renovate/gradle bash -l -c \"./gradlew --init-script renovate-plugin.gradle renovate\"\n\nFAILURE: Build failed with an exception.\n\n* Where:\nBuild file '/mnt/renovate/gh/DomiCraftFR/DomiDependencies/DomiDependencies-fabric/build.gradle.kts' line: 1\n\n* What went wrong:\nAn exception occurred applying plugin request [id: 'fabric-loom', version: '0.8-SNAPSHOT']\n> Failed to apply plugin 'fabric-loom'.\n   > You are using an outdated version of Java (11). Java 16 or higher is required.\n\n* Try:\nRun with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.\n\n* Get more help at https://help.gradle.org\n\nBUILD FAILED in 7m 15s\n"
}
```

</details>

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [x] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

Reproduction repository: https://github.com/DomiCraftFR/RenovateTest

<details><summary>Click me to see logs on the reproduction repository</summary>

```
WARN: Gradle extraction failed
{
  "errMessage": "Command failed: docker run --rm --name=renovate_gradle --label=renovate_child -v \"/mnt/renovate/gh/DomiCraftFR/RenovateTest\":\"/mnt/renovate/gh/DomiCraftFR/RenovateTest\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -e GRADLE_OPTS -w \"/mnt/renovate/gh/DomiCraftFR/RenovateTest\" docker.io/renovate/gradle bash -l -c \"./gradlew --init-script renovate-plugin.gradle renovate\"\n\nFAILURE: Build failed with an exception.\n\n* Where:\nBuild file '/mnt/renovate/gh/DomiCraftFR/RenovateTest/build.gradle.kts' line: 1\n\n* What went wrong:\nAn exception occurred applying plugin request [id: 'fabric-loom', version: '0.8-SNAPSHOT']\n> Failed to apply plugin 'fabric-loom'.\n   > You are using an outdated version of Java (11). Java 16 or higher is required.\n\n* Try:\nRun with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.\n\n* Get more help at https://help.gradle.org\n\nBUILD FAILED in 3m 58s\n"
}
```

**Comment by rarkins** [comment]
Can you try using the preset ":switchToGradleLite"?

**Reply by Aurelien30000** [reply]
Thank you very much!

**Comment by viceice** [comment, marked as answer]
`gradle` manager is deprecated and we can't update java. so please try the new `gradle-lite` manager with adding `:switchToGradleLite` preset. So please use the preset because of #10091

https://docs.renovatebot.com/presets-default/#switchtogradlelite

**Reply by Aurelien30000** [reply]
Thank you very much!
